File: src/main/java/me/markoutte/sandbox/profiling/CreateDirectories.java
package me.markoutte.sandbox.profiling;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public class CreateDirectories {

    public static void main(String[] args) throws IOException {
        Path path = Paths.get("dir");
        for (int i = 0; i < 100_000; i++) {
            Files.createDirectories(path);
        }
    }

}

File: src/main/java/me/markoutte/sandbox/profiling/CreateDirectories.kt
package me.markoutte.sandbox.profiling

import java.nio.file.Files
import java.nio.file.Paths
import java.util.concurrent.TimeUnit
import kotlin.system.measureNanoTime

fun main() = measureAndPrint {
    val path = Paths.get("dir")
    for (i in 0..99999) {
//        if (!Files.exists(path)) {
            Files.createDirectories(path)
//        }
    }
}

fun measureAndPrint(block: () -> Unit) {
    val nanos = measureNanoTime(block)
    val millis = TimeUnit.NANOSECONDS.toMillis(nanos)
    println("Total time $millis ms")
}